pkg/controller/migmigration/task.go

```diff
@@ -1,148 +1,58 @@
 package migmigration
 
 import (
-	"fmt"
 	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
 	"github.com/go-logr/logr"
-	velero "github.com/heptio/velero/pkg/apis/velero/v1"
+	"github.com/pkg/errors"
 	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
 )
 
-// Annotation Keys
-const MigQuiesceAnnotationKey = "openshift.io/migrate-quiesce-pods"
-
 // Phases
 const (
-	Created                 = ""
-	Started                 = "Started"
-	WaitOnResticRestart     = "WaitOnResticRestart"
-	ResticRestartCompleted  = "ResticRestartCompleted"
-	BackupStarted           = "BackupStarted"
-	BackupCompleted         = "BackupCompleted"
-	BackupFailed            = "BackupFailed"
-	WaitOnBackupReplication = "WaitOnBackupReplication"
-	BackupReplicated        = "BackupReplicated"
-	RestoreStarted          = "RestoreStarted"
-	RestoreCompleted        = "RestoreCompleted"
-	RestoreFailed           = "RestoreFailed"
-	Completed               = "Completed"
+	Created                       = ""
+	Started                       = "Started"
+	Prepare                       = "Prepare"
+	EnsureInitialBackup           = "EnsureInitialBackup"
+	InitialBackupCreated          = "InitialBackupCreated"
+	InitialBackupFailed           = "InitialBackupFailed"
+	AnnotateResources             = "AnnotateResources"
+	EnsureStagePods               = "EnsureStagePods"
+	StagePodsCreated              = "StagePodsCreated"
+	RestartRestic                 = "RestartRestic"
+	ResticRestarted               = "ResticRestarted"
+	QuiesceApplications           = "QuiesceApplications"
+	EnsureStageBackup             = "EnsureStageBackup"
+	StageBackupCreated            = "StageBackupCreated"
+	StageBackupFailed             = "StageBackupFailed"
+	EnsureInitialBackupReplicated = "EnsureInitialBackupReplicated"
+	EnsureStageBackupReplicated   = "EnsureStageBackupReplicated"
+	EnsureStageRestore            = "EnsureStageRestore"
+	StageRestoreCreated           = "StageRestoreCreated"
+	StageRestoreFailed            = "StageRestoreFailed"
+	EnsureFinalRestore            = "EnsureFinalRestore"
+	FinalRestoreCreated           = "FinalRestoreCreated"
+	FinalRestoreFailed            = "FinalRestoreFailed"
+	EnsureStagePodsDeleted        = "EnsureStagePodsDeleted"
+	EnsureAnnotationsDeleted      = "EnsureAnnotationsDeleted"
+	Completed                     = "Completed"
 )
 
-var PhaseOrder = map[string]int{
-	Created:                 00, // 0-499 normal
-	Started:                 1,
-	WaitOnResticRestart:     10,
-	ResticRestartCompleted:  11,
-	BackupStarted:           20,
-	BackupCompleted:         21,
-	WaitOnBackupReplication: 30,
-	BackupReplicated:        31,
-	RestoreStarted:          40,
-	RestoreCompleted:        41,
-	BackupFailed:            501, // 500-999 errors
-	RestoreFailed:           510,
-	Completed:               1000, // Succeeded
```

Comment:
Is dropping all of the PhaseOrder stuff intentional?

Reply:
Yes.
Since phases are not repeated, the ordering is no longer useful.

Reply:
Interesting point, didn't realize we could take out the ordering due to this. Seems like a good simplification.